internal/csi-addons/rbd/replication.go

```diff
@@ -270,24 +277,25 @@ func (rs *ReplicationServer) EnableVolumeReplication(ctx context.Context,
 	}
 	defer rs.VolumeLocks.Release(volumeID)
 
-	rbdVol, err := corerbd.GenVolFromVolID(ctx, volumeID, cr, req.GetSecrets())
-	defer func() {
-		if rbdVol != nil {
-			rbdVol.Destroy(ctx)
-		}
-	}()
+	mgr := rbd.NewManager(rs.csiID, req.GetParameters(), req.GetSecrets())
+	defer mgr.Destroy(ctx)
+	var mirror types.Mirror
+
+	rbdVol, err := mgr.GetVolumeByID(ctx, volumeID)
 	if err != nil {
 		switch {
 		case errors.Is(err, corerbd.ErrImageNotFound):
 			err = status.Errorf(codes.NotFound, "volume %s not found", volumeID)
 		case errors.Is(err, util.ErrPoolNotFound):
-			err = status.Errorf(codes.NotFound, "pool %s not found for %s", rbdVol.Pool, volumeID)
+			err = status.Errorf(codes.NotFound, "pool %s not found for %s", rbdVol.GetPoolName(), volumeID)
 		default:
 			err = status.Errorf(codes.Internal, err.Error())
 		}
 
 		return nil, err
 	}
+	mirror = rbdVol
```

Comment:
Would it make sense to have `rbdVol.GetMirror()` that validates everything required for the new Mirror type?

Reply:
@nixpanic can you explain a bit more on this one?

Reply:
updated the code, PTAL and see if this matches your expectation